lib/ansible/plugins/connection/vmware_tools.py

```diff
@@ -0,0 +1,297 @@
+"""Docstring for public module."""
+from __future__ import absolute_import, division, print_function
+
+import re
+from os.path import dirname, exists, getsize
+from socket import gaierror
+from ssl import SSLEOFError, SSLError
+from time import sleep
+
+import requests
+import urllib3
+from pyVim.connect import Disconnect, SmartConnect, SmartConnectNoSSL
+from pyVmomi import vim
+
+from ansible.errors import AnsibleError, AnsibleFileNotFound
+from ansible.module_utils._text import to_bytes, to_native
+from ansible.plugins.connection import ConnectionBase
+from ansible.utils.path import makedirs_safe
+
+__metaclass__ = type
+
+DOCUMENTATION = """
+    author: Deric Crago <deric.crago@gmail.com>
+    connection: vmware_tools
+    short_description: Execute modules via VMware Tools.
+    description:
+      - Execute modules via VMware Tools.
+    version_added: "2.8"
+    requirements:
+      - pyvmomi (python library)
+    options:
+      connection_address:
+        description:
+          - Address for the connection
+        vars:
+          - name: ansible_vmware_tools_connection_address
+        required: True
+      connection_username:
+        description:
+          - Username for the connection
+        vars:
+          - name: ansible_vmware_tools_connection_username
+        required: True
+      connection_password:
+        description:
+          - Password for the connection
+        vars:
+          - name: ansible_vmware_tools_connection_password
+        required: True
+      connection_verify_ssl:
+        description:
+          - Verify SSL for the connection
+        vars:
+          - name: ansible_vmware_tools_connection_verify_ssl
+        default: True
+        type: bool
+      connection_ignore_ssl_warnings:
+        description:
+          - Ignore SSL warnings for the connection
+        vars:
+          - name: ansible_vmware_tools_connection_ignore_ssl_warnings
+        default: False
+        type: bool
+      vm_path:
+        description:
+          - VM path relative to vCenter.
+          - Example: C(Datacenter/vm/Discovered virtual machine/testVM) (Needs to include C(vm) between the Datacenter and the rest of the VM path.)
+        vars:
+          - name: ansible_vmware_tools_vm_path
+        required: True
+      vm_username:
+        description:
+          - VM username.
+        vars:
+          - name: ansible_vmware_tools_vm_username
+        required: True
+      vm_password:
+        description:
+          - VM password.
+        vars:
+          - name: ansible_vmware_tools_vm_password
+        required: True
+      exec_command_sleep_interval:
+        description:
+          - exec command sleep interval in seconds.
+        vars:
+          - name: ansible_vmware_tools_exec_command_sleep_interval
+        default: 5
+        type: integer
+      file_chunk_size:
+        description:
+          - File chunk size.
```

Comment:
Add note saying this is applicable to put/get file method.

Reply:
Done.